### ISSUE TITLE
How to setup backups manually

### ISSUE THREAD

**Issue by GeorgeBellTMH** [aws-amplify/docs#2708]
It would be good to document how to setup comprehensive backups for amplify until there is a way to do it from the cli.

How to backup:
- S3
- Dynamo
- Cognito
- others???

How to restore:
- S3
- Dynamo
- Cognito
- others???

**Comment by GeorgeBellTMH** [comment]
https://github.com/aws-amplify/amplify-cli/issues/5972

**Comment by GeorgeBellTMH** [comment]
Seems important...

**Comment by swatipraskar** [comment]
Yes it’s important. Any news..For any serious App, you cannot go live without backup and DR features..

**Comment by droidgeek1** [comment]
I

> Seems important...

I second that it's important for any commercial application to have a solution to this problem. I am wondering why there isn't any activity on this? Is there another way that we don't know?

**Comment by droidgeek1** [comment]
I would add functions to the list! Along with sound sound Disaster recovery design

**Comment by rachelnabors** [comment]
Thanks so much for letting us know what you need! We just published a blog post on this topic: https://aws-preview.aka.amazon.com/blogs/storage/automate-backups-for-aws-amplify-graphql-backends-with-aws-backup/

**Comment by GeorgeBellTMH** [comment]
I'll just add that having now tried to actually backup and restore a cognito instance it is actually almost impossible...

**Comment by GeorgeBellTMH** [comment]
1. If you are restoring to a new instance there are all sorts of gotchas (including needing to setup SES, getting out of sandbox mode)
2. The CSV import in cognito has no export function...
3. The CSV import does not handle groups
4. It seems to send an email to each user asking them to reset via pin...(which eventually fails if 1 is not complete)
5. It's not clear when importing if there are setup problems with custom. fields or other associated setting differences

**Comment by GeorgeBellTMH** [comment]
Backup doesn't seem to be available out of the box with cognito...so you are at the mercy of API's and non AWS developers....

**Comment by kevinschwarz** [comment]
I just published a blog in the AWS Storage channel last week that provides the definition for a CDK custom resource to setup backups with AWS Backup. Take a look and see if this solves for your need.

https://aws.amazon.com/blogs/storage/automate-backups-for-aws-amplify-graphql-backends-with-aws-backup/
